src/Agent/NewRelic.Api.Agent/NewRelic.cs

```diff
@@ -605,7 +745,14 @@ public static void SetUserParameters(string? userName, string? accountName, stri
         [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
         public static void RecordCustomEvent(string eventType, IEnumerable<KeyValuePair<string, object>> attributes)
         {
-            System.Diagnostics.Trace.WriteLine(string.Format("NewRelic.RecordCustomEvent({0},{1})", eventType, attributes));
+            try
+            {
+                System.Diagnostics.Trace.WriteLine(string.Format("NewRelic.RecordCustomEvent({0},{1})", nameof(eventType), nameof(attributes)));
+            }
+            catch
+            {
+                // Swallow any exception thrown from here
+            }
```

Comment:
```suggestion
            System.Diagnostics.Trace.WriteLine(string.Format("NewRelic.RecordCustomEvent({0},{1})", nameof(eventType), nameof(attributes)));
```

Reply:
All the methods should be safe without the `try/catch` now that we're not attempting to `ToString` the parameters.

For example, this `RecordCustomEvent` method can be problematic if the underlying type's `ToString` method is not thread-safe when `attributes.ToString()` is called. The type would be something like:

```csharp
class MyEnumerable : IEnumerable<KeyValuePair<string, object>>
{
    // Performs work that is not thread-safe
    public override string ToString() ...
}
```

`Dictionary<string, object>` is such a type. https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.

Reply:
I don't think we positively established that the reason the customer (or any customer in general) might experience an exception from these tracing calls can only be a non-threadsafe `.ToString()`.  I'd prefer the belt-and-suspenders approach for maximum safety.